app/com/lynxanalytics/biggraph/frontend_operations/VertexAttributeOperations.scala

```diff
@@ -7,17 +7,87 @@ import com.lynxanalytics.biggraph.graph_api.Scripting._
 import com.lynxanalytics.biggraph.graph_operations
 import com.lynxanalytics.biggraph.graph_util.Scripting._
 import com.lynxanalytics.biggraph.controllers._
-import com.lynxanalytics.biggraph.graph_api.Attribute
-import com.lynxanalytics.biggraph.graph_api.Scalar
+import com.lynxanalytics.biggraph.graph_api._
+import com.lynxanalytics.biggraph.controllers.OperationParams._
 
+abstract class UseTableAsAttributeOperation(context: Operation.Context, ops: ProjectOperations)
+  extends ProjectOutputOperation(context) {
+  import Operation.Implicits._
+  override lazy val project = projectInput("graph")
+  lazy val table = tableLikeInput("attributes").asProject
+  def projectAttributes: StateMapHolder[Attribute[_]]
+  def projectIdSet: VertexSet
+  def tableAttributes = table.vertexAttributes
+  val projectAttrNames = projectAttributes.iterator.map(_._1).toList.sorted
+
+  params ++= List(
+    Choice("id_attr", "ID attribute",
+      options = FEOption.unset +: FEOption.list(projectAttrNames)),
+    Choice("id_column", "ID column",
+      options = FEOption.unset +: table.vertexAttrList),
+    Param("prefix", "Name prefix for the imported attributes"),
+    Choice("unique_keys", "Assert unique ID attribute values", options = FEOption.boolsDefaultFalse),
+    Choice("if_exists", "What happens if an attribute already exists",
+      options = FEOption.list(
+        "Overwrite from the table", "Keep the graph's version", "Use the table's version",
+        "They must match", "Disallow this")))
```

Comment:
I think there is an option missing: keep the graph version whenever defined but backfill from table if not defined in graph.

The right wording is very hard here.... It's intrinsically a bit complex, as conflict resolution can happen on two levels. Maybe a bit better to use instead of your "Overwrite from the table" and "They must match" these three:
"Merge: on conflicting value keep the graph's version"
"Merge: on conflicting value overwrite the with table's version"
"Merge: error on any conflicting value"
Or too long?

I think merge is a good word to put in here somehow, as what happens here in two cases is exactly what "Merge two ... attributes" does.

Reply:
> I think there is an option missing: keep the graph version whenever defined but backfill from table if not defined in graph.

Okay, why not.

> The right wording is very hard here.... It's intrinsically a bit complex, as conflict resolution can happen on two levels. Maybe a bit better to use instead of your "Overwrite from the table" and "They must match" these three:
> "Merge: on conflicting value keep the graph's version"
> "Merge: on conflicting value overwrite the with table's version"
> "Merge: error on any conflicting value"
> Or too long?

Thanks! I've used a similar wording, just a bit shorter. (`Merge, prefer the graph's version`, `Merge, prefer the table's version`, `Merge, report error on conflict`.)

Reply:
:+1: